xenmgr/Vm/Config.hs

```diff
@@ -686,9 +700,9 @@ nicSpec cfg amt eth0Mac nic networkDomID =
                 | Just mac <- eth0Mac, amt == True  = ["mac=" ++ unswizzleMac mac]
       -- Otherwise we do not touch the VM mac and let xenvm choose
                 | otherwise                         = [ ]
-      nicType   = if (vmcfgStubdom cfg) then ["type=ioemu"] else ["type=vif"]
-      modelType | Just model <- nicdefModel nic = if (vmcfgStubdom cfg) then ["model="++model] else []
-                | otherwise                     = if (vmcfgStubdom cfg) then ["model=e1000"] else []
+      nicType   = if isHvm cfg then ["type=ioemu"] else ["type=vif"]
+      modelType | Just model <- nicdefModel nic = if isHvm cfg then ["model="++model] else []
+                | otherwise                     = if isHvm cfg then ["model=e1000"] else []
```

Comment:
This assumes all HVMs have a stubdom, which is not true. It breaks the stubdom-less case by added emulated NICs to stubdom-less guests, causing hangs.

Reply:
Okay, I can drop 1f92cecedce973ff86f04501ce0ca29a3677c415 which introduced this change.  I wasn't totally sure about it.  So an stubdom-less HVM will not get presented an emulated NIC at all since QEMU in Dom0 cannot network it properly.  But a stubdom-less HVM can still use PV drivers to NDVM.

What "hangs" are you seeing?  Hangs for actual networking, or something else?

Thanks for taking a look at this.

Reply:
It'd be fine to overload this `if` to handle the PVH case, but here it was completely repurposed :)
When I start a Win10 guest with tools and without a stubdom, it hangs on the Welcome screen, probably waiting for the emulated NICs to show up or something.

Reply:
Yes, you're right. I'll have to redo it to check both hvm and stubdom.